File: app/src/main/java/github/umer0586/util/JsonUtil.java
package github.umer0586.util;


import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;
import java.util.List;

public class JsonUtil {


    private static ObjectMapper objectMapper = new ObjectMapper();


    public static String toJSON(Object object)
    {
        String json = "";

        try {
            json = objectMapper.writer().writeValueAsString(object);
        } catch (IOException e) {
            e.printStackTrace();
        }

        return json;
    }

    // Maps string elements in JSON array to Java list e.g [a,b,c]
    public static List<String> readJSONArray(String jsonArrayString)
    {
        try
        {
            return objectMapper.readValue(jsonArrayString,new TypeReference<List<String>>(){});

        } catch (IOException e)
        {
            e.printStackTrace();
        }
        return null;
    }



}

File: app/src/main/java/github/umer0586/sensorserver/util/JsonUtil.kt
package github.umer0586.sensorserver.util

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import java.io.IOException

object JsonUtil
{


    private val objectMapper = ObjectMapper()
    fun toJSON(`object`: Any?): String
    {
        var json = ""
        try
        {
            json = objectMapper.writer().writeValueAsString(`object`)
        }
        catch (e: IOException)
        {
            e.printStackTrace()
        }
        return json
    }

    // Maps string elements in JSON array to Java list e.g [a,b,c]
    fun readJSONArray(jsonArrayString: String?): List<String>?
    {
        try
        {
            return objectMapper.readValue<List<String>>(jsonArrayString, object :
                TypeReference<List<String>?>()
            {})
        }
        catch (e: IOException)
        {
            e.printStackTrace()
        }
        return null
    }
}